cmd/nginx-ingress/main.go

```diff
@@ -656,6 +663,22 @@ func createPlusAndLatencyCollectors(
 	return plusCollector, syslogListener, lc
 }
 
+func createHealthProbeEndpoint(kubeClient *kubernetes.Clientset, plusClient *client.NginxClient, cnf *configs.Configurator) {
+	var serviceInsightSecret *api_v1.Secret
+	var err error
+
```

Comment:
We could exit out early here if `*enableServiceInsight` is not set, instead of first checking the secret and then checking if it's set before we trigger the goroutine

Reply:
thanks! applied